tests/conftest.py

```diff
@@ -622,18 +642,31 @@ def xmlrpc(self, path, method, *args):
 
 
 @pytest.fixture
-def webtest(app_config):
-    # TODO: Ensure that we have per test isolation of the database level
-    #       changes. This probably involves flushing the database or something
-    #       between test cases to wipe any committed changes.
-
+def webtest(app_config_dbsession_from_env, db_session):
```

Comment:
There's an issue left here:

`db_session` already relies on `app_config`, so we need to ensure that pytest only needs to resolve a single fixture, not both `app_config_dbsession_from_env` and `app_config`.

Reply:
The above issue is solved in https://github.com/benjaoming/warehouse/compare/issue_15634_test_user_profile-investigation...benjaoming:warehouse:issue_15634_test_user_profile-investigation-refactor-dbsession?expand=1

@miketheman if you're happy to review with the above approach, I'll pull it in and make it ready :blush: 

Otherwise, please feel free to suggest an alternative. I'm not saying that this is the best minimal+correct approach... but more that I didn't have any other sudden ideas while working on it.

Reply:
@benjaoming That looks good to me!

Reply:
Great, I've pushed the approach to this PR branch :+1: